### PR TITLE
CPE: stop using deprecated CPEs, split the remap YAML into sections

### PR DESCRIPTION
**Pull request by tsellers-r7**
## Description
This PR addresses an issue brought up by @p0lr in Issue #343. In that issue @p0lr points out that we're using deprecated CPE values and that the NIST database that we're already using indicates that the values are deprecated.

I've adjusted the matching logic in `update_cpes.py` so that these deprecated CPEs aren't added when we build our list of NIST issued CPEs.

I've also restructured `cpe-remap.yaml` so that we remap values are specific to a CPE type (`a`, `o`, `h`). This will allow us to remap certain older records that use the same `.product` value but for which the CPE would be different between `service.product` and `os.product`.

For example in the fingerprint below we couldn't have created a remap for `ILOM` under `oracle` because it would have then generated (or tried to generate) the same CPE for both `hw.product` and `os.product`.

```xml
<fingerprint pattern="^(?:Oracle\(R\) )?Integrated Lights Out Manager$">
    <description>Oracle iLOM</description>
    <example>Oracle(R) Integrated Lights Out Manager</example>
    <example>Integrated Lights Out Manager</example>
    <param pos="0" name="hw.device" value="Lights Out Management"/>
    <param pos="0" name="hw.vendor" value="Oracle"/>
    <param pos="0" name="hw.family" value="ILOM"/>
    <param pos="0" name="hw.product" value="ILOM"/>
    <param pos="0" name="os.device" value="Lights Out Management"/>
    <param pos="0" name="os.vendor" value="Oracle"/>
    <param pos="0" name="os.family" value="ILOM"/>
    <param pos="0" name="os.product" value="ILOM"/>
  </fingerprint>
```

With the new system we can remap them independently which will help us generate CPEs for existing values without having to change them.

## Motivation and Context
Correctness of results.

## How Has This Been Tested?
Execution of `cpe-remap.yaml`, rspec, etc.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: identifiers/service_product.txt [hunk 1]
```diff
@@ -277,7 +278,7 @@ MetaDirectory Server
 Metabase
 Metasploit
 MiniDLNA
-MiniUPnP
+MiniUPnPd
```
Comment (tsellers-r7) [inline identifiers/service_product.txt:281]: There is actually a `MiniUPnPd` (daemon) and `MiniUPnPc` (client) so I changed this product value.

File: xml/html_title.xml [hunk 2]
```diff
@@ -44,6 +44,7 @@
     <param pos="0" name="service.vendor" value="Amazon"/>
     <param pos="0" name="service.family" value="CloudFront"/>
     <param pos="0" name="service.product" value="CloudFront Load Balancer"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:amazon:amazon_cloudfront:-"/>
   </fingerprint>
```
Comment (tsellers-r7) [inline xml/html_title.xml:48]: This as well as the new CPE for Akamai Ghost and a couple of AWS services is a result of requesting NIST to issue them. Thanks NIST!

File: xml/html_title.xml [hunk 3]
```diff
@@ -1695,10 +1710,11 @@
...
 
   <fingerprint pattern="^Login - OpenStack Dashboard$">
-    <description>OpenStack Dashboard</description>
+    <description>OpenStack Horizon Dashboard</description>
     <example>Login - OpenStack Dashboard</example>
     <param pos="0" name="service.vendor" value="OpenStack"/>
-    <param pos="0" name="service.product" value="Dashboard"/>
+    <param pos="0" name="service.product" value="Horizon"/>
```
Comment (tsellers-r7) [inline xml/html_title.xml:1716]: Incorrect product name but calling it out here in case anyone depends on it.

File: xml/html_title.xml [hunk 4]
```diff
@@ -2492,7 +2526,7 @@
     <param pos="0" name="os.device" value="Lights Out Management"/>
     <param pos="0" name="os.vendor" value="Oracle"/>
     <param pos="0" name="os.family" value="ILOM"/>
-    <param pos="0" name="os.product" value="Integrated Lights Out Manager firmware"/>
+    <param pos="0" name="os.product" value="ILOM"/>
```
Comment (tsellers-r7) [inline xml/html_title.xml:2529]: @hdm  - Returning this value to what it was earlier this week. You'd asked about the need to add ` firmware` to the end of the value in that PR.  I didn't like it at the time but I didn't have decent options.  With the current PR this is now trivial.

File: xml/http_servers.xml [hunk 5]
```diff
@@ -1364,13 +1364,13 @@
     <param pos="0" name="service.vendor" value="HP"/>
     <param pos="0" name="service.product" value="iLO"/>
     <param pos="0" name="service.family" value="iLO"/>
-    <param pos="0" name="service.cpe23" value="cpe:/a:hp:integrated_lights_out:-"/>
```
Comment (tsellers-r7) [inline xml/http_servers.xml:1367]: This was deprecated w/o a replacement.

File: xml/x509_subjects.xml [hunk 6]
```diff
@@ -1195,6 +1199,7 @@
     <example>CN=a248.e.akamai.net,O=Akamai Technologies\, Inc.,L=Cambridge,ST=Massachusetts,C=US</example>
     <param pos="0" name="service.vendor" value="Akamai"/>
     <param pos="0" name="service.product" value="GHost"/>
+    <param pos="0" name="service.cpe23" value="cpe:/a:akamai:akamaighost:-"/>
```
Comment (tsellers-r7) [inline xml/x509_subjects.xml:1202]: This entry alone will result in CPEs for 9.5+ million endpoints..

**Comment by tsellers-r7** [comment]
CC @hdm  @p0lr

File: xml/html_title.xml (hunk 3 above)
Comment (rkirk-r7) [reply xml/html_title.xml:1716]: Nexpose side does not depend on these

**Review by rkirk-r7 — APPROVED** [review]

**Pull request merged**